### PR TITLE
Dedicated static function to parse out mutator name

### PR DESCRIPTION
**Pull request by sanmai**
This PR:

- [x] Adds a dedicated static function to parse out mutator name
- [x] Covered by tests

### CODE REVIEW

File: src/Mutator/MutatorFactory.php [hunk 1]
```diff
@@ -92,4 +92,11 @@ public function create(array $resolvedMutators): array
 
         return $mutators;
     }
+
+    public static function getMutatorNameForClassName(string $className): string
+    {
+        $parts = explode('\\', $className);
+
+        return (string) end($parts);
+    }
```
Comment (sanmai) [inline src/Mutator/MutatorFactory.php:101]: Closest place where `Mutator::getName()` is used.

**Review by maks-rafalko — APPROVED** [review]

**Comment by theofidry** [comment]
Looks a bit weird to me that it's the factory responsible for giving the mutator name (by default)

**Comment by sanmai** [comment]
It is. Should be easy to refactor. What do you have in mind?

**Comment by theofidry** [comment]
What was wrong with how it was before? I saw there was a bunch of PRs so maybe I missed the overall picture

**Comment by sanmai** [comment]
It was only an instance method before (still is), but sometimes you want to get a name for a mutator having only a class name for it.

**Comment by theofidry** [comment]
Do you have a place where this is the case? I think it would help to understand what's going on because indeed it's a bit tricky

**Comment by sanmai** [comment]
Sure, [right about here](https://github.com/infection/infection/commit/ee40e434a0b3ec8385e3d1d5591c2013b72a7720#diff-f20466d79712e5e751a8819581cc4787R270-R285).

**Comment by theofidry** [comment]
Replied there

**Pull request merged**